bundles/org.openhab.core.thing/src/test/java/org/openhab/core/thing/internal/profiles/ToggleProfileTest.java

```diff
@@ -0,0 +1,188 @@
+        verifyAction(profile, UnDefType.NULL, OnOffType.ON);
+        verifyAction(profile, HSBType.WHITE, OnOffType.OFF);
+        verifyAction(profile, HSBType.BLACK, OnOffType.ON);
+        verifyAction(profile, new HSBType("0,50,50"), OnOffType.OFF);
+    }
+
+    @Test
+    public void testRollershutterItem() {
+        prepareContextMock();
+        TriggerProfile profile = newToggleProfile(RAWBUTTON_TOGGLE_ROLLERSHUTTER);
+        verifyAction(profile, UnDefType.NULL, UpDownType.UP);
+        verifyAction(profile, UpDownType.UP, UpDownType.DOWN);
+        verifyAction(profile, UpDownType.DOWN, UpDownType.UP);
+
+        profile = newToggleProfile(BUTTON_TOGGLE_ROLLERSHUTTER);
+        verifyAction(profile, UnDefType.NULL, UpDownType.UP);
+        verifyAction(profile, UpDownType.UP, UpDownType.DOWN);
+        verifyAction(profile, UpDownType.DOWN, UpDownType.UP);
+    }
+
+    @Test
+    public void testPlayerItem() {
+        prepareContextMock();
+        TriggerProfile profile = newToggleProfile(RAWBUTTON_TOGGLE_PLAYER);
+        verifyAction(profile, UnDefType.NULL, PlayPauseType.PLAY);
+        verifyAction(profile, PlayPauseType.PLAY, PlayPauseType.PAUSE);
+        verifyAction(profile, PlayPauseType.PAUSE, PlayPauseType.PLAY);
+
+        profile = newToggleProfile(BUTTON_TOGGLE_PLAYER);
+        verifyAction(profile, UnDefType.NULL, PlayPauseType.PLAY);
+        verifyAction(profile, PlayPauseType.PLAY, PlayPauseType.PAUSE);
+        verifyAction(profile, PlayPauseType.PAUSE, PlayPauseType.PLAY);
+    }
+
+    @Test
+    public void testCorrectUserConfiguredEvent() {
+        ListAppender<ILoggingEvent> logAppender = createLogAppender();
+
+        initializeContextMock("RELEASED");
+        TriggerProfile profile = newToggleProfile(RAWBUTTON_TOGGLE_SWITCH);
+        assertEquals(0, logAppender.list.size());
+        verifyAction(profile, UnDefType.NULL, OnOffType.ON, "RELEASED");
+        verifyAction(profile, OnOffType.ON, OnOffType.OFF, "RELEASED");
+        verifyAction(profile, OnOffType.OFF, OnOffType.ON, "RELEASED");
+    }
+
+    @Test
+    public void testWrongUserConfiguredEvent() {
+        ListAppender<ILoggingEvent> logAppender = createLogAppender();
+        initializeContextMock("SHORT_PRESSED");
+        TriggerProfile profile = newToggleProfile(RAWBUTTON_TOGGLE_SWITCH);
+
+        assertEquals(Level.WARN, logAppender.list.get(0).getLevel());
+        verifyAction(profile, OnOffType.ON, OnOffType.OFF);
+    }
+
+    private void initializeContextMock(@Nullable String triggerEvent) {
+        Map<String, Object> params = triggerEvent == null ? Collections.emptyMap()
+                : Collections.singletonMap(ToggleProfile.EVENT_PARAM, triggerEvent);
+        when(contextMock.getConfiguration()).thenReturn(new Configuration(params));
+    }
+
+    private void prepareContextMock() {
+        initializeContextMock(null);
+    }
+
+    private ListAppender<ILoggingEvent> createLogAppender() {
```

Comment:
There is a method for that in `JavaTest`: `setupInterceptedLogger(Class<?> clazz, LogLevel minLogLevel)` and also assertions for no log (`assertNoLogMessage`) or log messages (`assertLogMessage(Class<?> clazz, LogLevel logLevel, String message)`.

Reply:
done